packages/types/index.ts

```diff
@@ -0,0 +1,161 @@
+import type { AsyncIterator } from 'asynciterator';
+import type { Map } from 'immutable';
+import type * as RDF from 'rdf-js';
+import type { Algebra } from 'sparqlalgebrajs';
+
+/**
```

Comment:
Just wondering if we want to split this up into separate files, not sure yet...
In any case, we can still do this in a non-breaking future change if needed.

Reply:
No strong opinion on this one.

Reply:
Actually, I would rather keep them in the same file until the latter becomes exceedingly long and unmaintainable.

Reply:
Sounds good to me.